visualization/include/visualization/render.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <string>
+
+#include "line.hpp"
+#include "rectangle.hpp"
+#include "circle.hpp"
+
+namespace visualization::render {
+
+    class Renderer {
```

Comment:
you need to add a virtual deconstructor like this:
```cpp
virtual ~Renderer() = default;
```
you can learn more why [here](https://stackoverflow.com/a/461224)